tracing/zipkin/README.md

```diff
@@ -36,131 +35,97 @@ The following services have been set-up to run:
-	defer cancel()
-
-	reply, err := client(ctx, param1, param2)
-	// do something with the response/error
-	...
-}
-```
+	// create a new span to record the resource interaction
+	span := opentracing.StartChildSpan(parentSpan, queryLabel)
 
-## Annotating Remote Resources
+	// span.kind "resource" triggers SA annotation
+	ext.SpanKind.Set(span, "resource")
 
-Next to the above shown examples of wiring server-side and client-side tracing
-middlewares, you can also span resources called from your service methods.
+	// this will label the span's service & hostPort (called Endpoint in Zipkin)
+	ext.PeerService.Set(span, serviceName)
+	ext.PeerHostname,Set(span, serviceHost)
+	ext.PeerPort.Set(span, servicePort)
 
-To do this, the service method needs to include a context parameter. From your
-endpoint wrapper you can inject the endpoint context which will hold the parent
-span already created by the server-side middleware. If the resource is a remote
-database you can use the `zipkin.ServerAddr` spanOption to identify the remote
-host:port and the display name of this resource.
+	// a Tag is the equivalent of a Zipkin Binary Annotation (key:value pair)
+	span.SetTag("query", query)
 
-```go
-type MyService struct {
-	// add a Zipkin Collector to your service implementation's properties.
-	Collector zipkin.Collector
-}
+	// a LogEvent is the equivalent of a Zipkin Annotation (timestamped)
+	span.LogEvent("query:start")
 
-// Example of the endpoint.Endpoint to service method wrapper, injecting the
-// context provided by the transport server.
-func makeComplexEndpoint() endpoint.Endpoint {
-	return func(ctx context.Context, request interface{}) (interface{}, error) {
-		req := request.(ComplexRequest)
-		v, err := svc.Complex(ctx, req.A, req.B)
-		return ComplexResponse{V: v, Err: err}, nil
-	}
-}
+	// do the actual query...
 
-// Complex is an example method of our service, displaying the tracing of a
-// remote database resource.
-func (s *MyService) Complex(ctx context.Context, A someType, B otherType) (returnType, error) {
-	// we've parsed the incoming parameters and now we need to query the database.
-	// we wish to include this action into our trace.
-	span, collect := zipkin.NewChildSpan(
-		ctx,
-		s.Collector,
-		"complexQuery",
-		zipkin.ServerAddr(
-			"mysql01.internal.net:3306",
-			"MySQL",
-		),
-	)
-	// you probably want to binary annotate your query
-	span.AnnotateBinary("query", "SELECT ... FROM ... WHERE ... ORDER BY ..."),
-	// annotate the start of the query
-	span.Annotate("complexQuery:start")
-	// do the query and handle resultset
-	...
-	// annotate we are done with the query
-	span.Annotate("complexQuery:end")
-	// maybe binary annotate some items returned by the resultset
-	...
-	// when done with all annotations, collect the span
-	collect()
+	// let's annotate the end...
+	span.LogEvent("query:end")
+
+	// we're done with the span. send it to the SpanRecorder for collection...
```

Comment:
This is how the Zipkin implementation works and that's the scope of this document, but I'll leave it out if it would confuse people.